torch/csrc/jit/source_range.h

```diff
@@ -1,23 +1,49 @@
 #pragma once
+#include <c10/util/Optional.h>
+#include <memory>
+
 #include <torch/csrc/jit/source_location.h>
 #include <torch/csrc/jit/assertions.h>
 
 
 namespace torch {
 namespace jit {
 
+struct SourceContext {
+  SourceContext(
+    c10::optional<std::shared_ptr<std::string>> file_name = c10::nullopt,
+    size_t line = 0
+  ) : file_name_(file_name), line_(line) {}
+
+  size_t line() const {
+    return line_;
+  }
+  c10::optional<std::shared_ptr<std::string>> file_name() const {
+    return file_name_;
+  }
+private:
+  c10::optional<std::shared_ptr<std::string>> file_name_;
+  size_t line_;
+};
+
 // a range of a shared string 'file_' with functions to help debug by highlight
 // that
 // range.
 struct SourceRange : public SourceLocation {
   SourceRange(
       std::shared_ptr<std::string> file_,
       size_t start_,
-      size_t end_)
-      : file_(std::move(file_)), start_(start_), end_(end_) {}
+      size_t end_,
+      c10::optional<std::shared_ptr<SourceContext>> context_ = c10::nullopt)
```

Comment:
This is completely redundant with the above parameters, so I can’t see why are we even adding this.